R/agg_scale.R

```diff
@@ -57,6 +55,53 @@
 #' @return \[`data.table()`\] with `id_cols` and `value_cols` columns for
 #'   requested aggregates or with scaled values.
 #'
+#' @section Severity Arguments:
+#' **`missing_dt_severity`**:
+#'
+#' Check for missing levels of `col_stem`, the variable being aggregated or
+#' scaled over.
+#' 1. `stop`: throw error (this is the default).
+#' 2. `warning` or `message`: throw warning/message and continue with
+#' aggregation/scaling where possible.
+#' 3. `none`: don't throw error or warning, continue with aggregation/scaling
+#' where possible.
+#' 4. `skip`: skip this check and continue with aggregation/scaling.
+#'
+#' **`present_agg_severity`** (`agg` only):
+#'
+#' Check for requested aggregates in `mapping` that are already present
+#' 1. `stop`: throw error (this is the default).
+#' 2. `warning` or `message`: throw warning/message, drop aggregates and continue
+#' with aggregation.
+#' 3. `none`: don't throw error or warning, drop aggregates and continue with
+#' aggregation.
+#' 4. `skip`: skip this check and add to the values already present for the
+#' aggregates.
+#'
+#' **`na_value_severity`**:
+#'
+#' Check for 'NA' values in the `value_cols`.
+#' 1. `stop`: throw error (this is the default).
+#' 2. `warning` or `message`: throw warning/message, drop missing values and
+#' continue with aggregation/scaling where possible (this likely will cause
+#' another error because of `missing_dt_severity`.
+#' 3. `none`: don't throw error or warning, drop missing values and continue
+#' with aggregation/scaling where possible (this likely will cause another error
+#' because of `missing_dt_severity`.
+#' 4. `skip`: skip this check and propagate `NA` values through
```

Comment:
What about a `na.rm = T` equivalent so that you aggregate anyway and just add `0` for missings? See the age `0-4` example I wrote out in this review.

Reply:
So this is similar to the test you requested below. 

You'd need to use `na_value_severity` and `missing_dt_severity`, so that the na values are dropped but we want to force aggregation/scaling.
```
    na_value_severity = "none",
    missing_dt_severity = "skip"
```

Reply:
Tried clarifying this 
```
#' 2. `warning` or `message`: throw warning/message, drop missing values and
#' continue with aggregation/scaling where possible (this likely will cause
#' another error because of `missing_dt_severity`, consider setting
#' `missing_dt_severity = "skip"` for functionality similiar to `na.rm = TRUE`).
#' 3. `none`: don't throw error or warning, drop missing values and continue
#' with aggregation/scaling where possible (this likely will cause another error
#' because of `missing_dt_severity`, consider setting
#' `missing_dt_severity = "skip"` for functionality similiar to `na.rm = TRUE`).
```